zio-blocks/src/main/scala/zio/blocks/schema/Optic.scala

```diff
@@ -110,82 +111,143 @@ sealed trait Lens[F[_, _], S, A] extends Optic[F, S, A] {
 object Lens {
   type Bound[S, A] = Lens[Binding, S, A]
 
-  def apply[F[_, _], S, A](parent: Reflect.Record[F, S], child: Term[F, S, A]): Lens[F, S, A] = new Field(parent, child)
-
-  def apply[F[_, _], S, T, A](first: Lens[F, S, T], second: Lens[F, T, A]): Lens[F, S, A] = new LensLens(first, second)
-
-  private case class Field[F[_, _], S, A](parent: Reflect.Record[F, S], child: Term[F, S, A])
-      extends Lens[F, S, A]
-      with Leaf[F, S, A] {
+  def apply[F[_, _], S, A](parent: Reflect.Record[F, S], child: Term[F, S, A]): Lens[F, S, A] = {
     require((parent ne null) && (child ne null))
+    new LensImpl(ArraySeq(parent), ArraySeq(child))
+  }
 
-    def structure: Reflect[F, S] = parent
-
-    def focus: Reflect[F, A] = child.value
+  def apply[F[_, _], S, T, A](first: Lens[F, S, T], second: Lens[F, T, A]): Lens[F, S, A] = {
+    val u1 = first.asInstanceOf[LensImpl[F, S, A]]
+    val u2 = second.asInstanceOf[LensImpl[F, S, A]]
+    new LensImpl(u1.parents ++ u2.parents, u1.childs ++ u2.childs)
+  }
 
-    private val register: Register[A] =
-      parent.registers(parent.fields.indexWhere(_.name == child.name)).asInstanceOf[Register[A]]
+  private case class LensImpl[F[_, _], S, A](
+    parents: ArraySeq[Reflect.Record[F, S]],
+    childs: ArraySeq[Term[F, S, A]]
+  ) extends Lens[F, S, A]
+      with Leaf[F, S, A] {
+    private[this] var bindings: Array[(Deconstructor[Any], Constructor[Any], Register[Any], RegisterOffset)] = null
```

Comment:
Instead of using `Tuple4` here, consider using a custom `case class` which is only used here. This will avoid boxing `RegisterOffset`.